examples/rabbit_fox_hawk.jl

```diff
@@ -0,0 +1,428 @@
+Rabbit(id, pos, energy) = Animal(id, pos, :rabbit, energy)
+Fox(id, pos, energy) = Animal(id, pos, :fox, energy)
+Hawk(id, pos, energy) = Animal(id, pos, :hawk, energy)
+norm(vec) = √sum(vec .^ 2)
+
+# The environment is generated from a heightmap: a 2D matrix, where each value denotes the
+# height of the terrain at that point. We segregate the model into 4 regions based on the
+# height:
+# - Anything below `water_level` is waster and cannot be walked on
+# - The region between `water_level` and `grass_level` is flatland, that can be walked on
+# - The part of the map between `grass_level` and `mountain_level` is too high for animals to
+#   walk over, but it can be flown over
+# - The terrain above `mountain_level` is completely inaccessible
+#
+# Grass is the food source for rabbits. It can grow anywhere from `water_level` to `grass_level`.
+# The spread of grass across the terrain is specified using a BitArray. A value of
+# 1 at a location indicates the presence of grass there, which can be consumed when it is eaten
+# by a rabbit. The probability of grass growing is proportional to how close it is to the water.
+#
+# The `initialize_model` function takes in the URL to our heightmap, the thresholds for the 4
+# regions, and some additional parameters for the model. It then creates and returns a model
+# with the specified heightmap and containing the specified number of rabbits, foxes and hawks.
+
+function initialize_model(
+    heightmap_url,
+    water_level = 8,
+    grass_level = 20,
+    mountain_level = 35;
+    n_rabbits = 160,  ## initial number of rabbits
+    n_foxes = 30,  ## initial number of foxes
+    n_hawks = 30,  ## initial number of hawks
+    Δe_grass = 25,  ## energy gained from eating grass
+    Δe_rabbit = 30,  ## energy gained from eating one rabbit
+    rabbit_repr = 0.06,  ## probability for a rabbit to (asexually) reproduce at any step
+    fox_repr = 0.03,  ## probability for a fox to (asexually) reproduce at any step
+    hawk_repr = 0.02, ## probability for a hawk to (asexually) reproduce at any step
+    rabbit_vision = 6,  ## how far rabbits can see grass and spot predators
+    fox_vision = 10,  ## how far foxes can see rabbits to hunt
+    hawk_vision = 15,  ## how far hawks can see rabbits to hunt
+    rabbit_speed = 1.3, ## movement speed of rabbits
+    fox_speed = 1.1,  ## movement speed of foxes
+    hawk_speed = 1.2, ## movement speed of hawks
+    regrowth_chance = 0.03,  ## probability that a patch of grass regrows at any step
+    dt = 0.1,   ## discrete timestep each iteration of the model
+    seed = 42,  ## seed for random number generator
+)
+
+    ## Download and load the heightmap. The grayscale value is converted to `Float64` and
+    ## scaled from 1 to 40
+    heightmap = floor.(Int, convert.(Float64, load(download(heightmap_url))) * 39) .+ 1
+    ## The x and y dimensions of the pathfinder are that of the heightmap
+    dims = (size(heightmap)..., 100)
```

Comment:
If you change the last value here to 50 and the z dimension in space to 50.0, then the plot looks more reasonable.